--- origsrc/evolution-sharp-0.21.1/configure.in	2010-01-25 03:29:09.965961100 -0600
+++ src/evolution-sharp-0.21.1/configure.in	2010-01-04 17:10:49.953632500 -0600
@@ -98,17 +98,28 @@ if test "x$GACUTIL" = "xno"; then
 	AC_MSG_ERROR([Your mono installation doesn't expose gacutil])
 fi
 
-if test `uname -s` = "Darwin"; then
-        LIB_PREFIX=
+case `uname -s` in
+    Darwin)
+        LIB_PREFIX=lib
+        LIB_MIDDLE=.
         LIB_SUFFIX=.dylib
-else
-        LIB_PREFIX=.so
+        ;;
+    CYGWIN*)
+        LIB_PREFIX=cyg
+        LIB_MIDDLE=-
+        LIB_SUFFIX=.dll
+        ;;
+    *)
+        LIB_PREFIX=lib
+        LIB_MIDDLE=.so.
         LIB_SUFFIX=
-fi
+        ;;
+esac
 
 AC_SUBST(CSC)
 AC_SUBST(GACUTIL)
 AC_SUBST(LIB_PREFIX)
+AC_SUBST(LIB_MIDDLE)
 AC_SUBST(LIB_SUFFIX)
 
 PKG_CHECK_MODULES([GLIB_SHARP],
--- origsrc/evolution-sharp-0.21.1/evolution/evolution-sharp.dll.config.in	2010-01-25 03:29:09.972961500 -0600
+++ src/evolution-sharp-0.21.1/evolution/evolution-sharp.dll.config.in	2010-01-25 03:30:12.895560500 -0600
@@ -1,7 +1,7 @@
 <configuration>
-  <dllmap dll="ebook" target="libebook@EDS_VERSION@@LIB_PREFIX@.@EBOOK_SOVERSION@@LIB_SUFFIX@"/>
-  <dllmap dll="ecal" target="libecal@EDS_VERSION@@LIB_PREFIX@.@ECAL_SOVERSION@@LIB_SUFFIX@"/>
-  <dllmap dll="edataserver" target="libedataserver@EDS_VERSION@@LIB_PREFIX@.@EDS_SOVERSION@@LIB_SUFFIX@"/>
-  <dllmap dll="ebackend" target="libebackend@EDS_VERSION@@LIB_PREFIX@.@EBACKEND_SOVERSION@@LIB_SUFFIX@"/>
-  <dllmap dll="edatacal" target="libedata-cal@EDS_VERSION@@LIB_PREFIX@.@EDATACAL_SOVERSION@@LIB_SUFFIX@"/>
+  <dllmap dll="ebook" target="@LIB_PREFIX@ebook@EDS_VERSION@@LIB_MIDDLE@@EBOOK_SOVERSION@@LIB_SUFFIX@"/>
+  <dllmap dll="ecal" target="@LIB_PREFIX@ecal@EDS_VERSION@@LIB_MIDDLE@@ECAL_SOVERSION@@LIB_SUFFIX@"/>
+  <dllmap dll="edataserver" target="@LIB_PREFIX@edataserver@EDS_VERSION@@LIB_MIDDLE@@EDS_SOVERSION@@LIB_SUFFIX@"/>
+  <dllmap dll="ebackend" target="@LIB_PREFIX@ebackend@EDS_VERSION@@LIB_MIDDLE@@EBACKEND_SOVERSION@@LIB_SUFFIX@"/>
+  <dllmap dll="edatacal" target="@LIB_PREFIX@edata-cal@EDS_VERSION@@LIB_MIDDLE@@EDATACAL_SOVERSION@@LIB_SUFFIX@"/>
 </configuration>
