--- origsrc/evolution-sharp-0.15.1/configure.in	2007-10-30 08:47:37.000000000 -0500
+++ src/evolution-sharp-0.15.1/configure.in	2007-12-14 07:24:46.927250000 -0600
@@ -5,6 +5,7 @@
 AM_MAINTAINER_MODE
 
 AC_PROG_INSTALL
+AM_DISABLE_STATIC
 AM_PROG_LIBTOOL
 
 AC_PATH_PROG(PKG_CONFIG, pkg-config, no)
@@ -85,17 +86,28 @@
 	AC_MSG_ERROR([Your mono installation doesn't expose gacutil])
 fi
 
-if test `uname -s` = "Darwin"; then
-        LIB_PREFIX=
+case `uname -s` in
+    Darwin)
+        LIB_PREFIX=lib
+        LIB_MIDDLE=.
         LIB_SUFFIX=.dylib
-else
-        LIB_PREFIX=.so
+        ;;
+    CYGWIN*)
+        LIB_PREFIX=cyg
+        LIB_MIDDLE=-
+        LIB_SUFFIX=.dll
+        ;;
+    *)
+        LIB_PREFIX=lib
+        LIB_MIDDLE=.so.
         LIB_SUFFIX=
-fi
+        ;;
+esac
 
 AC_SUBST(CSC)
 AC_SUBST(GACUTIL)
 AC_SUBST(LIB_PREFIX)
+AC_SUBST(LIB_MIDDLE)
 AC_SUBST(LIB_SUFFIX)
 
 PKG_CHECK_MODULES(GTK_SHARP, gtk-sharp-2.0 >= 2.4.0)
--- origsrc/evolution-sharp-0.15.1/evolution/evolution-sharp.dll.config.in	2007-06-19 21:17:29.000000000 -0500
+++ src/evolution-sharp-0.15.1/evolution/evolution-sharp.dll.config.in	2007-12-14 07:24:46.942875000 -0600
@@ -1,5 +1,5 @@
 <configuration>
-  <dllmap dll="ebook" target="libebook@EDS_VERSION@@LIB_PREFIX@.@EBOOK_SOVERSION@@LIB_SUFFIX@"/>
-  <dllmap dll="ecal" target="libecal@EDS_VERSION@@LIB_PREFIX@.@ECAL_SOVERSION@@LIB_SUFFIX@"/>
-  <dllmap dll="edataserver" target="libedataserver@EDS_VERSION@@LIB_PREFIX@.@EDS_SOVERSION@@LIB_SUFFIX@"/>
+  <dllmap dll="ebook" target="@LIB_PREFIX@ebook@EDS_VERSION@@LIB_MIDDLE@@EBOOK_SOVERSION@@LIB_SUFFIX@"/>
+  <dllmap dll="ecal" target="@LIB_PREFIX@ecal@EDS_VERSION@@LIB_MIDDLE@@ECAL_SOVERSION@@LIB_SUFFIX@"/>
+  <dllmap dll="edataserver" target="@LIB_PREFIX@edataserver@EDS_VERSION@@LIB_MIDDLE@@EDS_SOVERSION@@LIB_SUFFIX@"/>
 </configuration>
--- origsrc/evolution-sharp-0.15.1/glue/Makefile.am	2007-06-19 21:17:30.000000000 -0500
+++ src/evolution-sharp-0.15.1/glue/Makefile.am	2007-12-14 07:24:46.958500000 -0600
@@ -27,7 +27,7 @@
 	$(ECAL_GLUE_SOURCES) \
 	$(EBOOK_GLUE_SOURCES)
 
-libevolutionglue_la_LDFLAGS = -module -avoid-version -no-undefined
+libevolutionglue_la_LDFLAGS = -avoid-version -no-undefined
 
 libevolutionglue_la_LIBADD =		\
 	$(LIBECAL_LIBS)			\
